kratos/includes/gid_io.h

```diff
@@ -236,14 +293,14 @@ class KRATOS_API(KRATOS_CORE) GidIO : public GidIOBase
     }//SetUpMeshContainers
 
     /**
-     * creates the gauss point containers for all different element types.
-     * Note that the containers are not filled yet in here!
+     * @brief Creates the gauss point containers for all different element types.
+     * @note The containers are not filled yet in here!
      */
     virtual void SetUpGaussPointContainers()
     {
         //elements with 1 gauss point
         std::vector<int> gp_indices(1);
-        gp_indices[0] = 0;
+        gp_indices[0] ;
```

Comment:
Then, shouldn't we remove this line? In any case, though it is initialized to zero, I'd keep it as before for the sake of clarity (personal opinion).

Reply:
(same for the rest)

Reply:
Huum, this will not work, the constructor does not initialize by zero

Reply:
I tried in cpp shell and it initializes to zero :open_mouth: (I wasn't aware).

Reply:
Indeed, I've just checked and if initialization value is not provided, it calls the default constructor of the type.

Reply:
Beware, it is filled with whatever, sometimes 0s, sometimes first value in memory

Reply:
See the constructor we're using here
```c++
explicit vector(
    size_type count,
    const T& value = T(),
    const Allocator& alloc = Allocator());
```
it calls `T()` which in this case results in `int()` which returns 0.

In any case, this will be changes to
```c++
explicit vector(
    size_type count,
    const T& value,
    const Allocator& alloc = Allocator());
```
in c++14.